File: app/src/main/java/pl/tysia/maggwarehouse/Presentation/PresentationLogic/CatalogAdapter/ICatalogable.java
package pl.tysia.maggwarehouse.Presentation.PresentationLogic.CatalogAdapter;

import android.graphics.Bitmap;

import pl.tysia.maggwarehouse.Presentation.PresentationLogic.Filterer.IFilterable;

public interface ICatalogable extends IFilterable {
    String getTitle();
    String getShortDescription();
    boolean isMarked();

}

File: app/src/main/java/pl/tysia/maggwarehouse/Presentation/PresentationLogic/CatalogAdapter/ICatalogable.kt
package pl.tysia.maggwarehouse.Presentation.PresentationLogic.CatalogAdapter

import pl.tysia.maggwarehouse.Presentation.PresentationLogic.Filterer.IFilterable

public interface ICatalogable : IFilterable{
    abstract fun getTitle(): String
    abstract fun getShortDescription(): String
    abstract fun getImageResource(): Int

    abstract fun getQuantityOrdered(): Int
    abstract fun setQuantityOrdered(quantityOrdered: Int)
    abstract fun getQuantityTaken(): Int
    abstract fun setQuantityTaken(quantityTaken: Int)
}